alf/environments/adv_fetch_envs/adv_fetch_env.py

```diff
@@ -0,0 +1,118 @@
+# Copyright (c) 2022 Horizon Robotics and ALF Contributors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import numpy as np
+
+from gym.envs.robotics.fetch_env import FetchEnv
+from gym.envs.robotics import robot_env, rotations, utils
+
+
+class AdvFetchEnv(FetchEnv):
+    def __init__(
+            self,
+            model_path,
+            n_substeps,
+            gripper_extra_height,
+            block_gripper,
+            has_object,
+            target_in_the_air,
+            target_offset,
+            obj_range,
+            target_range,
+            distance_threshold,
+            initial_qpos,
+            reward_type,
+    ):
+        """Initializes a new Fetch environment. Almost the same with ``FetchEnv``.
+        # The only change is from 4 to 7 for ``n_actions`` where the extra 3 dims
+        represent 'xyz' euler angles rotated compared to the previous step.
```

Comment:
document the meaning of each dimension of the action. It seems that 'xyz' euler angles are not the last 3 dimensions.

Reply:
> document the meaning of each dimension of the action. It seems that 'xyz' euler angles are not the last 3 dimensions.

added